ci/install_docker.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+# Add Docker's official GPG key:
+apt-get update
+apt-get --yes install ca-certificates curl gnupg
+install -m 0755 -d /etc/apt/keyrings
+curl -fsSL https://download.docker.com/linux/debian/gpg | gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+chmod a+r /etc/apt/keyrings/docker.gpg
+
+# Add the repository to Apt sources:
+echo \
+  "deb [arch="$(dpkg --print-architecture)" signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/debian \
+  "$(. /etc/os-release && echo "$VERSION_CODENAME")" stable" | \
+  tee /etc/apt/sources.list.d/docker.list > /dev/null
+apt-get update
+
+apt-get --yes install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
```

Comment:
Maybe we can use already pre-made action for this such as: https://github.com/marketplace/actions/setup-docker ?

Reply:
That action hardcodes the usage of sudo (which is not present in the `rust:1.72` image). So to use it I would have to add sudo installation and setup which I don't think is worth it. I think I would prefer to keep the official way of installing docker as is (see https://docs.docker.com/engine/install/debian/#install-using-the-repository).